Tensile/TensileCreateLibrary.py

```diff
@@ -1109,6 +1109,47 @@ def param(key, value):
       
       param("best-solution", True)
 
+def generateLazyMasterFileList(masterFileList: List[Tuple[str, MasterSolutionLibrary]]) -> List[Tuple[str, MasterSolutionLibrary]]:
+    """ Generates a list of tuples that represent the name and the state associated with the lazy master libraries.
+
+    This function takes a list of MasterSolutionLibraries and traverses each lazy libraries.
+    It collects the items (i.e. the name and corresponding master file) and adds them to list
+    of master files.
+
+    Args:
+        masterLibraries: A list of name / master solution library pairs.
+    """
+    return masterFileList + list(itertools.chain.from_iterable((t for t in lib.lazyLibraries.items()) for _, lib in masterFileList))
+
+def generateMasterFileList(masterLibraries: dict, archs: List[str], lazy: bool) -> List[Tuple[str, MasterSolutionLibrary]]:
+    """ Generates a list of tuples that represent the name and the state associated with the architecture
+        specific master libraries.
+
+    This function takes a dictionary with keys corresponding to a target architecture and values 
+    corresponding to the master solution library for that architecture. The function generates a
+    tuple consisting of a MasterSolutionLibrary and the associated name.
+
+    Args:
+        masterLibraries: A dictionary of architecture name / master solution library pairs.
+        archs: A list of supported architectures.
+        lazy: If True, add lazy library master files.
+    """
+    baseName = "TensileLibrary_lazy_" if lazy else "TensileLibrary_"
+    result = [(baseName + arch, masterLibrary) for arch, masterLibrary in masterLibraries.items() if arch in archs]
+    return generateLazyMasterFileList(result) if lazy else result
```

Comment:
Not opposed to this, but the readability would be better IMO if it read like 
```python
    return result + generateLazyMasterFileList(result) if lazy else result
```
and the extension of the existing masterFileList isn't subsumed within the function call, thoughts?

Reply:
I am good with that change